db/migrate/20220926164441_add_slug_to_article.rb

```diff
@@ -0,0 +1,21 @@
+class AddSlugToArticle < ActiveRecord::Migration[6.1]
+  def up
+    add_column :articles, :slug, :string, default: ''
+
+    update_past_articles_with_slug
+
+    add_index :articles, :slug
+    change_column_null(:articles, :slug, false)
+  end
+
+  def down
+    remove_column(:articles, :slug)
+  end
+
+  def update_past_articles_with_slug
+    Article.all.each do |article|
+      slug = article.title.underscore.parameterize(separator: '_')
```

Comment:
We might have to add a fallback condition here for the cases where someone created articles with the same title. 
We could catch the error and try updating the slug with numerical increments in the end.

Reply:
The separator should be `-` not `_`